### ISSUE TITLE
Rotate labels when many categorical labels on axis

### ISSUE THREAD

**Issue by morganmendis** [DistrictDataLabs/yellowbrick#188]
Need label rotation to angled text when there are too many categorical labels on an axis (example: ClassBalance report)

This was identified by neal@nhumphrey.com in their user study response.

**Comment by bbengfort** [comment]
@rebeccabilbro ping PyLadies who were working on this project.

**Comment by Aylr** [comment]
This is also an issue when labels are long in Parallel Coordinates.
<img width="546" alt="screen shot 2017-09-07 at 11 26 22 pm" src="https://user-images.githubusercontent.com/928247/30197292-12f47c94-9424-11e7-9b7a-de27a2ef7096.png">

**Comment by jaydee829** [comment]
This hasn't been closed so I thought I would add a workaround to this 2-year old issue. If you directly access the visualizer axes, you can call tick_params as follows:

`visualizer.ax.tick_params(axis='x', labelrotation=90.)`

Here is the output on 110 features ;-)
![xtick_rotation](https://user-images.githubusercontent.com/42184471/58110068-cc1c8500-7bbc-11e9-8e06-9157831a6fae.png)

**Comment by bbengfort** [comment]
@jaydee829 great workaround thank you for sharing!

**Comment by GegznaV** [comment]
I use Jupyter Notebook in VSCode and the proposed workaround does not work unless `show=False` is set, e.g:

```python
visualizer = parallel_coordinates(X_train, y_train, normalize="standard", show=False)
visualizer.ax.tick_params(axis='x', labelrotation=45)
```

**Comment by GegznaV** [comment]
Is there a way to achieve the same effect with 
```python
visualizer.ax.tick_params(axis='x', labelrotation=45)
```
as with `ha='right'` in:
```python
import matplotlib.pyplot as plt

plt.xticks(rotation=45, ha='right');
```
Argument `ha` is not implemented in `visualizer.ax.tick_params()`. Is there another way to right-align labels using `visualizer.ax.tick_params()`?

**Comment by bbengfort** [comment]
@GegznaV thank you so much for using Yellowbrick. @pdamodaran is going to take a look at your question and respond to you as soon as she can.

**Comment by pdamodaran** [comment]
Hi @GegznaV - based on my investigation, I believe you should be able to achieve the desired result without having to set `show=False`.  Here is a screenshot of an example:

![image](https://github.com/DistrictDataLabs/yellowbrick/assets/7106134/977165b6-dbaa-45da-a809-09ae6a82998d)

If you want to use the `ha` parameter, here is another option, but this results in a warning and the resulting visualization is slightly different:

<img width="1007" alt="image" src="https://github.com/DistrictDataLabs/yellowbrick/assets/7106134/938cf7e0-54fa-4391-af8e-41d4db3b166a">

Looking into the warning message, it appears that the community seems to be in favor of using `ax.tick_params` but there is no option to use `ha` as you have discovered.

Let me know if this answers your question or if you have any additional questions.  Thanks for using Yellowbrick!

**Comment by GegznaV** [comment]
Both examples work. As I understand, using class instead of function has more flexibility. Thank you.

I add copyable code in case anybody needs to reproduce the examples.

```python
from yellowbrick.datasets import load_credit 
from yellowbrick.features import ParallelCoordinates
X, y = load_credit()

# Instantiate the visualizer
visualizer = ParallelCoordinates(
    features=X.columns, sample=0.05, shuffle=True, normalize="standard"
)
# Fit and transform the data to the visualizer 
visualizer.fit_transform(X, y)
visualizer.ax.set_xticklabels(visualizer.ax.get_xticklabels(), rotation=45, ha="right") 
visualizer.show()
```

**Comment by pdamodaran** [comment]
You're welcome - glad this worked out for you!  And thanks for adding a code snippet for reference.  I am going to close this issue.  Thanks for using Yellowbrick!
